lib/pathutils.c

```diff
@@ -36,3 +37,25 @@ is_file_regular(const char *filename)
 {
   return g_file_test(filename, G_FILE_TEST_EXISTS) && g_file_test(filename, G_FILE_TEST_IS_REGULAR);
 };
+
+gboolean
+is_file_device(const gchar *name)
+{
+  struct stat st;
+
+  if (stat(name, &st) >= 0)
+    return S_ISCHR(st.st_mode) || S_ISBLK(st.st_mode);
+  else
+    return FALSE;
+}
+
+gboolean
+is_file_socket(const gchar *name)
+{
+  struct stat st;
+
+  if (stat(name, &st) >= 0)
+    return S_ISSOCK(st.st_mode);
+  else
+    return TRUE;
```

Comment:
The `return TRUE` is mandatory because it is valid scenario if the socket is not exist.